python/ray/data/_internal/planner/map_batches.py

```diff
@@ -105,6 +106,12 @@ def process_next_batch(batch: DataBatch) -> Iterator[Block]:
                 else:
                     raise e from None
 
+        try:
+            first_block = next(blocks)
```

Comment:
This would hold the an extra block in-memory during execution, right? To avoid the increased memory overhead, can we create the corresponding output block builder here?

Reply:
No I don't think it will hold any extra memory? We're not copying the block anywhere.

Reply:
hmm, but in this case, the first block is always living in object store memory, until the `map_batches` is finished, right? because we hold the reference here.